include/zephyr/net/wifi_mgmt.h

```diff
@@ -503,6 +503,22 @@ enum wifi_mgmt_op {
 	WIFI_MGMT_SET = 1,
 };
 
+#define MAX_REG_CHNL_NUM  42
+
+/** Regulatory per channel atributes */
+struct wifi_reg_chnl_info {
+	/** center frequency in MHz */
+	unsigned int center_frequency;
+	/** maximum transmission power (in dBm) */
+	unsigned int max_power;
+	/** Particular channel is supported or not */
+	bool supported;
+	/** Particular channel is supports passive scanning or not */
+	bool passive_channel;
+	/** Particular channel is dfs or not */
+	bool dfs;
```

Comment:
Just wondering if we could make this struct smaller.

Could we use `short int` here? That could save 4 bytes per element.
Also booleans could be set as bit fields, which could save 2 bytes.
If we pack the struct, then that could save extra 1 byte.

Reply:
Updated the structure elements to reduce the size. current size is about 136 bytes.